CppGraphFiles/NumberOfShortestPathsCircle.cpp

```diff
@@ -0,0 +1,96 @@
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <limits>
+#include <queue>
+#include <list>
+
+typedef std::vector<std::list<std::pair<int, int>>> adjacency_list;
+
+bool VertexInRnge(int vertex, int verticesCount)
+{
+    return vertex >= 0 && vertex < verticesCount;
+}
+
+adjacency_list ReadGraph(std::ifstream& in)
+{
+    int vertices, edges;
+
+    in >> vertices >> edges;
+
+    if (vertices < 0 || edges < 0)
+    {
+        throw std::logic_error("Wrong value for vertices or edges");
+    }
+
+    adjacency_list graph(vertices);
+
+    for (int i = 0; i < edges; ++i)
+    {
+        int vertex1, vertex2, weight;
+
+        in >> vertex1 >> vertex2 >> weight;
+
+        if (VertexInRnge(vertex1, vertices) && VertexInRnge(vertex2, vertices) && weight >= 0)
+        {
+            graph[vertex1].push_back(std::make_pair(vertex2, weight));
+            graph[vertex2].push_back(std::make_pair(vertex1, weight));
+        }
+        else
+        {
+            throw std::logic_error("Wrong edge or wrong weight");
+        }
+    }
+
+    return graph;
+}
+
+int main(int argc, char** argv)
```

Comment:
What does this file?

Reply:
I changed that files name, now it's name is lengthOfShortestPathCircle.cpp.
It calculates with the formula the length of the shortest path in circle graph.